wp/lib/bap_wp/src/precondition.ml

```diff
@@ -709,7 +709,13 @@ let visit_jmp (env : Env.t) (post : Constr.t) (jmp : Jmp.t) : Constr.t * Env.t =
               (Expr.to_string cond_val) (hooks_to_string hooks);
             let cond_size = BV.get_size (Expr.get_sort cond_val) in
             let false_cond = Bool.mk_eq ctx cond_val (z3_expr_zero ctx cond_size) in
-            let ite = Constr.mk_ite jmp (Bool.mk_not ctx false_cond) l_pre post in
+            let is_unconditional = Expr.equal cond_val (z3_expr_one ctx cond_size) in
```

Comment:
I'm not crazy about this: we're turning a BAP expression into a Z3 expression which we're then comparing as a Z3 expression to the bit vector one.

We should try to just stay withing BAP land, it should be more robust and less confusing.

Reply:
`cond` is within scope as well, so it seems equally convenient.

Reply:
Was able to check if `cond` is unconditional by checking it with `Word.is_one`.